index-handlers/src/main/java/no/sikt/nva/nvi/index/utils/InstitutionReportGenerator.java

```diff
@@ -0,0 +1,129 @@
+    }
+
+    private Stream<List<String>> generateDataRows(NviCandidateIndexDocument candidate) {
+        return generateReportRowsForCandidate(candidate).stream();
+    }
+
+    private List<NviCandidateIndexDocument> fetchNviCandidates() {
+        return attempt(() -> searchClient.search(buildSearchRequest())).orElseThrow()
+                   .hits()
+                   .hits()
+                   .stream()
+                   .map(Hit::source)
+                   .toList();
+    }
+
+    private CandidateSearchParameters buildSearchRequest() {
+        return CandidateSearchParameters.builder()
+                   .withYear(year)
+                   .withTopLevelCristinOrg(topLevelOrganization)
+                   .build();
+    }
+
+    private List<List<String>> generateReportRowsForCandidate(NviCandidateIndexDocument candidate) {
+        return attempt(() -> generateReportRowsForContributorAffiliations(candidate)).map(Stream::toList)
+                   .orElseThrow(failure -> {
+                       logFailure(failure.getException(), candidate);
+                       return (RuntimeException) failure.getException();
+                   });
+    }
+
+    private Stream<List<String>> generateReportRowsForContributorAffiliations(NviCandidateIndexDocument candidate) {
+        return candidate.getNviContributors().stream()
+                   .flatMap(nviContributor -> generateRowsForContributorAffiliations(candidate, nviContributor));
+    }
+
+    private void logFailure(Exception exception, NviCandidateIndexDocument candidate) {
+        logger.error("Failed to generate report lines for candidate: {}. Error {}", candidate.id(),
+                     getStackTrace(exception));
+    }
+
+    private Stream<List<String>> generateRowsForContributorAffiliations(NviCandidateIndexDocument candidate,
+                                                                        NviContributor nviContributor) {
+        return nviContributor.getAffiliationsPartOfOrEqualTo(topLevelOrganization)
+                   .map(affiliation -> generateRow(candidate, nviContributor, affiliation));
+    }
+
+    private List<String> generateRow(NviCandidateIndexDocument candidate, NviContributor nviContributor,
+                                     NviOrganization affiliation) {
+        var row = new ArrayList<String>();
+        row.add(candidate.getReportingPeriodYear());
+        row.add(candidate.getPublicationIdentifier());
+        row.add(candidate.getPublicationDateYear());
+        row.add(getApprovalStatus(candidate));
+        row.add(candidate.getPublicationInstanceType());
+        row.add(nviContributor.id());
+        row.add(affiliation.getInstitutionIdentifier());
+        row.add(affiliation.getFacultyIdentifier());
+        row.add(affiliation.getDepartmentIdentifier());
+        row.add(affiliation.getGroupIdentifier());
+        row.add(nviContributor.name());
+        row.add(nviContributor.name());
```

Comment:
Is the duplicate value here intentional? The headers seem to be LAST_NAME and FIRST_NAME